From ee0064ad89a50ce07ffec8d132b42c98e1edd52a Mon Sep 17 00:00:00 2001
From: Tyler Johnson <tylerjohnson3208@gmail.com>
Date: Thu, 5 Mar 2026 17:47:45 -0800
Subject: [PATCH] Allow inclusion of local RELEASE and CONFIG_SITE

---
 configure/CONFIG_SITE | 2 ++
 configure/RELEASE     | 2 ++
 2 files changed, 4 insertions(+)

diff --git a/configure/CONFIG_SITE b/configure/CONFIG_SITE
index 8d75463..ceb9443 100644
--- a/configure/CONFIG_SITE
+++ b/configure/CONFIG_SITE
@@ -18,3 +18,5 @@ RE2C = re2c$(HOSTEXE)
 
 # Parser generator
 LEMON = $(call PATH_FILTER,$(INSTALL_HOST_BIN)/lemon$(HOSTEXE))
+
+-include $(TOP)/configure/CONFIG_SITE.local
diff --git a/configure/RELEASE b/configure/RELEASE
index 2541246..7230e4b 100644
--- a/configure/RELEASE
+++ b/configure/RELEASE
@@ -5,3 +5,5 @@
 
 EPICS_BASE=/opt/Epics/R3.14.12/base/3-14-12-2-1
 # EPICS_BASE=/opt/Epics/R3.15/base/head
+
+-include $(TOP)/configure/RELEASE.local
